crates/sdk/src/prover.rs

```diff
@@ -0,0 +1,40 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use sp1_core_executor::{ExecutionError, ExecutionReport};
+use sp1_core_machine::io::SP1Stdin;
+use sp1_primitives::io::SP1PublicValues;
+use sp1_prover::{SP1ProvingKey, SP1VerifyingKey};
+
+use crate::{opts::ProofOpts, proof::SP1ProofWithPublicValues, provers::SP1VerificationError};
+
+#[async_trait]
+pub trait Prover: Send + Sync {
```

Comment:
Id imagine this just needs to be `Sync`?

Reply:
updated